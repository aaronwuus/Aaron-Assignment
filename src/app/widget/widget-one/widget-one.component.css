.widgetOne {
    width: 300px;
    height: 200px;
    margin-right: 20px;
}
.subBtn {
    margin-top: 20px;
}