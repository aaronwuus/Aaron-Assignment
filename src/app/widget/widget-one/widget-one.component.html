<div class="card border-info mb-3 widgetOne" style="max-width: 18rem;">
    <div class="card-header">Widget One</div>
    <div class="card-body text-info ">
    <input class="form-control" name="widgetOne" type="text" placeholder="" #data />
    <button type="button" class="btn btn-primary subBtn" (click)="send(data)">Submit</button>
    </div>
</div>