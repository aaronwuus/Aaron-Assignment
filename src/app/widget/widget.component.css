.wrap {
    text-align: center; 
    margin: auto;
    position: relative;
    height: 500px;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display:table;
    top:50%;
    transform:translateY(50%);
}
.row-wrap {
    text-align: center; 
    position: relative;
    left:50%;
    transform:translateX(-25%); 
}