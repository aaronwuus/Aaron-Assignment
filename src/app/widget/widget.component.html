<div class="container wrap">
    <div class="row row-wrap">
        
            <app-widget-one></app-widget-one>

            <app-widget-two></app-widget-two>

    </div>  
</div>