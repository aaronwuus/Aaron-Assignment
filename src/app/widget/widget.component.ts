import { NeoBrowseService } from './../services/neo-browse.service';
import { NavbarService } from './../services/navbar.service';
import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-widget',
  templateUrl: './widget.component.html',
  styleUrls: ['./widget.component.css']
})
export class WidgetComponent implements OnInit {

  constructor(private navbarService: NavbarService, private neoBrowseService: NeoBrowseService) { }

  ngOnInit() {
    this.navbarService.show();
    this.neoBrowseService.navPage.next(1);
  }

}
