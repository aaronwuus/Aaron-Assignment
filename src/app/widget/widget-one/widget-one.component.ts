import { WidgetService } from './../../services/widget.service';
import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-widget-one',
  templateUrl: './widget-one.component.html',
  styleUrls: ['./widget-one.component.css']
})
export class WidgetOneComponent implements OnInit {
  constructor(private widgetService: WidgetService) { }

  ngOnInit() {
  }
  send(data) {
    this.widgetService.subject.next(data.value);
  }
}
