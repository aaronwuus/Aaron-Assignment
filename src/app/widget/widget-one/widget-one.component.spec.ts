import { WidgetComponent } from './../widget.component';
import { HttpClientModule } from '@angular/common/http';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { WidgetOneComponent } from './widget-one.component';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('WidgetOneComponent', () => {
  let component: WidgetOneComponent;
  let fixture: ComponentFixture<WidgetOneComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ WidgetOneComponent, WidgetComponent ],
      schemas: [NO_ERRORS_SCHEMA],
      imports: [HttpClientModule],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(WidgetOneComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toMatchSnapshot();
  });
});
