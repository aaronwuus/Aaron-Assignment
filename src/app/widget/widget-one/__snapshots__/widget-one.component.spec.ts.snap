// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetOneComponent should create 1`] = `
WidgetOneComponent {
  "widgetService": WidgetService {
    "subject": BehaviorSubject {
      "_isScalar": false,
      "_value": "",
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
  },
}
`;
